### ISSUE TITLE
Aid / Help other characters

### ISSUE THREAD

**Issue by johannes-z** [ehrpg/qstar#9]
Define rules for aiding/helping other characters in their task.

**Comment by n0la** [comment]
The following questions arise:

*Q: Do the main task roller and the helper need to be close in terms of skill?*

Yes and no. If I do programming work, and you are not a programmer you can't help much. Except by feeding me coffee. However if I build a small house, you could still help by carrying stuff or doing other menial labour even if you don't have the skills (mason, joiner etc.) required.

*Q: Are there restrictions?*

There must be some. Imagine that I shoot someone, and you from the other cover shoot at him too. Are you helping me kill him? I'd doubt it. Either I hit him, or you. But if I move from cover, and you do suppressive fire are you helping me? Yes, but is that within the definition of aiding me? Some things are just easier in groups. Even police show up in pairs in interrogations because it's easier to intimidate/bluff someone if there are more of you. But where do two separate rolls end and aiding start?

*Q: How effective should aiding be?*

I'd argue not at all. I believe that aiders should contribute modifiers, reduced time or add other benefits (less materials used, better quality of produced item) instead of dices. Adding dices would be so powerful, that aiding would be something that is done all the time for the sheer benefit it provides. Also it makes sense to allow the GM to be able to restrict the number of aiders. Only two people could perhaps pick a lock together without getting in each others way, while it's totally fine if five people service a spaceship together.

**Comment by johannes-z** [comment]
> Q: Do the main task roller and the helper need to be close in terms of skill?

You could probably argue, that easier tasks (that are below a certain DC threshold) can be aided, while more difficult tasks require a certain skill level. That skill level doesn't have to correspond with the actual character doing the task. F.e. a very good programmer could do the task easily. Someone else is helping you, who isn't as good, but he is still a programmer. He could do the task as well, but just not as good/fast. If he wasn't able to contribute at all, then aiding makes little sense (maybe a GM discretion?). Anyway, even if you were to program something difficult, even talking about the problems that arise often helps, whether or not someone else actually contributes to solving them...

> Q: Are there restrictions?

I'd recon that tasks that directly affect someone else shouldn't be aid-able. Meaning: If two characters were to intimidate someone, both would need to roll a separate roll. If somebody is trying to decrypt a secret message, that somebody can be aided. This would basically split tasks/actions into two categories. Also, there just might be tasks, where aiding does not make sense (whether gameplay-mechanically or realistically).

> Q: How effective should aiding be?

I'd say, that separate tasks should **always** be better, than helping someone who is already an expert in a skill. Aiding should create a mechanic, to:

1. Improve the chance of success for an expert (a little bit).
2. **Help someone bad at a certain skill to succeed.**

For 2. I'm thinking of a group of characters trying to climb up a steep wall (classic example). Some of them are maybe skilled at climbing or don't have an penalties, whilst others may. In the real world this would require the "aiders" to be skilled at climbing, to effectively help them (as in give them tips where to tread, secure ropes, encourage them, find spots where they could rest their legs and arms). Of course, the more skilled a character is, the better he can help somebody else.

**Comment by n0la** [comment]
> F.e. a very good programmer could do the task easily. Someone else is helping you, who isn't as good, but he is still a programmer. He could do the task as well, but just not as good/fast. If he wasn't able to contribute at all, then aiding makes little sense (maybe a GM discretion?).

Yes GM discretion is needed. Tasks done with skill checks, especially when in crisis or under time pressure, don't mean much. If you have thirty minutes to find and fix some weird bug in a PHP/Javascript app you wrote or the ship blows up, I would not be of much use. Even as a programmer I know neither the languages, nor the architecture of your app. I might even make things worse. The same with climbing.

> For 2. I'm thinking of a group of characters trying to climb up a steep wall (classic example). Some of them are maybe skilled at climbing or don't have an penalties, whilst others may. In the real world this would require the "aiders" to be skilled at climbing, to effectively help them (as in give them tips where to tread, secure ropes, encourage them, find spots where they could rest their legs and arms). Of course, the more skilled a character is, the better he can help somebody else.

A non sporty person (like me) could never climb a decent cliff, regardless of whether there's a professional climber talking me through it. Or a non pilot might never be able to dock or land a space ship, regardless of whether the tower control guy talks him through it.

> I'd say, that separate tasks should always be better, than helping someone who is already an expert in a skill. Aiding should create a mechanic, to:
>* Improve the chance of success for an expert (a little bit).
>* Help someone bad at a certain skill to succeed.

I agree. When is aiding used?

* When you have an expert and he needs to do something that absolutely should succeed.
* When you have a bunch of people doing something they have no clue of. Like two people trying to figure out the first time how to assemble IKEA furniture.
* When time is of the essence.
* When one player has a skill that sort of complements another. Say you need to blow up a structure, and the engineering guy and the explosive guy work together to find out were to best place the C4.
* Or when you have a player who has nothing to do.

> I'd recon that tasks that directly affect someone else shouldn't be aid-able. Meaning: If two characters were to intimidate someone, both would need to roll a separate roll.

What if one person holds the enemy, and I hit him with the bat? The fact that he is pinned makes it easier for me to hit him, even though the enemy might be good enough to dodge or block. Like all those wimpy mafia bosses in movies, who'd never win in a straight fight, suddenly landing punches against a superior opponent just because those are being pinned by boss's body guard gorillas. Since it is a competition (his roll against yours) you can't adjust difficulties. So aiding would be the needed to make it easier here for the one that has the benefit.

The general idea would then be:

## Aiding

Upon GM's discretion, it is possible for one or more players to aid another in an upcoming task (henceforth called the *main task*). The aiders must declare, and be able to do so, before the main task is attempted (i.e. rolled), and must roll a skill check (consuming an action) during their turn. Every success translates into a tangible benefit for the main task, such as:

* One additional modifier for each success, to be used in the roll for the main task.
* Improved output of the main task (less resources used, less time consumed, more health healed etc.)

The actual benefits are up to the GM. More often than not players will state the use of their aid, for example to minimise resources used (such as time, building materials). Players should declare beforehand what their help is trying to achieve, so that the GM can design the benefit accordingly. GM's may also demand that a task is not doable alone, such as lifting or moving something heavy that cannot be moved or lifted alone without aid or tools.

**Comment by johannes-z** [comment]
> What if one person holds the enemy, and I hit him with the bat? 

The enemy being hold is a condition due to the person holding him. Indirectly that means he is aiding the attacker, but gameplay mechanically it's not. 

I think the rules for **Aiding** are good as is. I'm not sure if we should leave them that "undefined" in the case of balancing. As we both know, GMs have a broad perception of how they think things work. Of course, GMs should always be able to bend the rules, but the rules themselves shouldnt require them to do so... if that makes sense.

**Comment by n0la** [comment]
> I think the rules for Aiding are good as is

Alright.

> I'm not sure if we should leave them that "undefined" in the case of balancing. As we both know, GMs have a broad perception of how they think things work. Of course, GMs should always be able to bend the rules, but the rules themselves shouldnt require them to do so... if that makes sense.

What do you wish to fix? Or what do you think are missing?

**Comment by johannes-z** [comment]
> What do you wish to fix? Or what do you think are missing?

In f.e. D&D 3.5e there is a very clear rule as to what aiding does and how it works. You need a certain skill rank, then you can aid someone else, granting them a +2 stackable bonus.

In our case it feels like... it could be whatever the GM wants to, and the player might really know beforehand if his idea of aiding would be sensible. Of course, this doesn't have to be too strict (that's the point of a pen&paper system anyway, right). Right now, the rules don't state how high the `additional modifier` should be, or what the improvement of the outcome should be. This leaves a huge room of trial & error for the GM - which might be good or bad, but I assume the latter is the case...

**Comment by n0la** [comment]
> Right now, the rules don't state how high the additional modifier should be

They do. **One** additional modifier **per success** of those that aid. Example:

Main task needs DC 6, guy that rolls has two dices and a modifier of +1. One guy helps him with one dice:

Helper rolls  **7**, a success, thus the main guy gets +1 modifier

Main task guy rolls **4** **2** He has +1 from ability and +1 from the success earlier, so +2 modifier. He spends those on the four and thus makes the DC of 6.

> or what the improvement of the outcome should be.

That's because the length of a task, or the resources involved are up to the GM, since we do not have rules for such things. It's easier to say, for example, *"for everyone that aids her task takes a quarter as long"*, or *"for every aider you use half as much gp while crafting"* if we had specific rules on how those things actually work (like 3.5E does). We don't, and I think that's actually a good thing. Thus the GM has come up with something in his head, and thus knows best on how give out benefits. 

> This leaves a huge room of trial & error for the GM

In 3.5e whenever we did something that's not as clearly covered by the rules (say base building) we bullshitted resource costs and time it took. If someone helped, we also bullshitted those rules. It works decently well, and there's always a possibility to argue such things over. But true, those rules require a bit more experience by the GM.
